docs/okeps/okep-5233-pre-assigned-pod-net-id.md

```diff
@@ -0,0 +1,455 @@
++ // The maximal exceptions allowed is 25.
++ // The format should match standard CIDR notation (for example, "10.128.0.0/16").
++ // This field must be omitted if `subnets` is unset or `ipam.mode` is `Disabled`.
++ // +optional
++ // +kubebuilder:validation:MinItems=1
++ // +kubebuilder:validation:MaxItems=25
++ ExcludeSubnets []CIDR `json:"excludeSubnets,omitempty"`
+
+// JoinSubnets are used inside the OVN network topology.
+//
+// Dual-stack clusters may set 2 subnets (one for each IP family), otherwise only 1 subnet is allowed.
+// This field is only allowed for "Primary" network.
+// It is not recommended to set this field without explicit need and understanding of the OVN network topology.
+// When omitted, the platform will choose a reasonable default which is subject to change over time.
+//
+// +optional
+JoinSubnets DualStackCIDRs `json:"joinSubnets,omitempty"`
+
++ // infrastructureSubnets specifies a list of internal CIDR ranges that OVN-Kubernetes will reserve for internal network infrastructure.
++ // Any IP addresses within these ranges cannot be assigned to workloads.
++ // When omitted, OVN-Kubernetes will automatically allocate IP addresses from `subnets` for its infrastructure needs.
++ // When `excludeSubnets` is also specified the CIDRs cannot overlap.
++ // When `defaultGatewayIPs` is also specified  the default gateway IPs cannot be a part of any of the CIDRs.
++ // Each item should be in range of the specified CIDR(s) in `subnets`.
++ // The maximal exceptions allowed is 10.
++ // The format should match standard CIDR notation (for example, "10.128.0.0/16").
++ // This field must be omitted if `subnets` is unset or `ipam.mode` is `Disabled`.
++ // +optional
++ // +kubebuilder:validation:MinItems=1
++ // +kubebuilder:validation:MaxItems=10
++ InfrastructureSubnets []CIDR `json:"infrastructureSubnets,omitempty"`
+
++ // defaultGatewayIPs specifies the default gateway IP used in the internal OVN topology.
++ //
++ // Dual-stack clusters may set 2 IPs (one for each IP family), otherwise only 1 IP is allowed.
++ // This field is only allowed for "Primary" network.
++ // It is not recommended to set this field without explicit need and understanding of the OVN network topology.
++ // When omitted, the first IP address from the pod network subnet is used.
++ //
++ // +optional
++ DefaultGatewayIPs DualStackIPs `json:"defaultGatewayIPs,omitempty"`
+
+// IPAM section contains IPAM-related configuration for the network.
+// +optional
+IPAM *IPAMConfig `json:"ipam,omitempty"`
+}
+
+// +kubebuilder:validation:XValidation:rule="isIP(self)", message="IP is invalid"
+type IP string
+
+// +kubebuilder:validation:MinItems=1
+// +kubebuilder:validation:MaxItems=2
+// +kubebuilder:validation:XValidation:rule="size(self) != 2 || !isIP(self[0]) || !isIP(self[1]) || ip(self[0]).family() != ip(self[1]).family()", message="When 2 IPs are set, they must be from different IP families"
+type DualStackIPs []IP
+
+```
```

Comment:
should there be a section for "NAD API changes?"

are these fields also going to be added to the NAD? Because in downstream OEP: https://github.com/openshift/enhancements/pull/1793/files#diff-ef7ea92bb7f0198174529787f912ec7898b84fa65e3e1b102ce30ed76735ab43R250 I see L250 @maiqueb suggests making changes to the internal structs to hold these field values which I assume we are going to also allow uses to use this via the NAD API?